### ISSUE TITLE
JSON parsing error for rendered output

### ISSUE THREAD

**Issue by rmzelle** [larsgw/citation.js#142]
I'm having trouble using citation.js to convert CSL JSON to rendered citations. Converting to bibtex works as expected.

Based on https://runkit.com/larsgw/591b5651bd9b40001113931c, I get the following results:

```js
const Cite = require("citation-js")

const data = new Cite([{"id":78,"type":"article-journal","title":"Identification of Positive Regulators of the Yeast Fps1 Glycerol Channel","container-title":"PLoS Genet","page":"e1000738","volume":"5","issue":"11","source":"PLoS Genet","journalAbbreviation":"PLoS Genet","author":[{"family":"Beese","given":"Sara E."},{"family":"Negishi","given":"Takahiro"},{"family":"Levin","given":"David E."}]}])

data.get({
  type: 'html',
  style: 'bibtex'
})
```

gives

```
@article{BeeseIdentification,

    author={Sara E. Beese and Takahiro Negishi and David E. Levin},
    journal={PLoS Genet},
    issue=11,
    pages={e1000738},
    title={{Identification of Positive Regulators of the Yeast Fps1 Glycerol Channel}},
    volume=5,

}
```

In contrast,

```js
const Cite = require("citation-js")

const data = new Cite([{"id":78,"type":"article-journal","title":"Identification of Positive Regulators of the Yeast Fps1 Glycerol Channel","container-title":"PLoS Genet","page":"e1000738","volume":"5","issue":"11","source":"PLoS Genet","journalAbbreviation":"PLoS Genet","author":[{"family":"Beese","given":"Sara E."},{"family":"Negishi","given":"Takahiro"},{"family":"Levin","given":"David E."}]}])

data.get({
  type: 'html',
  style: 'citation-apa'
})
```

gives the error

> SyntaxError: Unexpected token u in JSON at position 0

(see https://runkit.com/5af4a2c58a83410012ddbf47/5af4a2d65bba2900126994f2#)

If I use `new Cite('Q21972834')` instead of using raw CSL-JSON as input both work. Any idea what's going on here?

**Comment by larsgw** [comment]
I'm not entirely sure what the problem is, but it seems to be fixed in the `v0.4.0-1` beta (which I see doesn't have a bundle yet). If you're okay with using this version (it has some new APIs, but should be backwards compatible), I can add the bundles to the archive branch. Alternatively, they are also available on the site:

* https://citation.js.org/src/citation.js
* https://citation.js.org/src/citation.min.js

**Comment by larsgw** [comment]
Okay, it seems to be caused by the ID being a number, while citeproc-js casts it to a string, and then gets `undefined`, which it doesn't handle particularly graceful. In the new versions of citation-js, strings and other types of IDs are always cast to a string (which wasn't really my intention, as far as I can recall).

I could backport a fix if needed.

**Comment by rmzelle** [comment]
Thanks for sorting this out!

No need to backport this for me. I probably want to be on 0.4.x anyway for
the (experimental) RIS support.

I prefer to use a tagged release URL (to the minimized library), so it
would be handy to have a 0.4.x release.

**Comment by larsgw** [comment]
I released the `v0.4.0-3` beta with RIS support, the CDN should load up soon:

* https://cdn.rawgit.com/larsgw/citation.js/archive/citation.js/citation-0.4.0-3.js
* https://cdn.rawgit.com/larsgw/citation.js/archive/citation.js/citation-0.4.0-3.min.js

**Comment by larsgw** [comment]
Here's a demo on how to use the RIS output BTW: https://runkit.com/larsgw/citation.js-ris-output-demo

Basically this:

```js
Cite(...).get({
  format: 'string',
  type: 'string',
  style: 'ris'
})
```

Or, with the new API:

```js
Cite(...).format('ris')
```

Additionally, you can pass parameters:

```js
Cite(...).format('ris', {
  type: 'text' // alternative: 'object' gives you an object representation
})
```
